[Bug]: missing find_package(libtrossen_arm REQUIRED) in the demos CMAKELISTS.txt

### What happened?

When I compile the c++ demos (via make) I get the following:

```bash
-- Generating done
-- Build files have been written to: /home/santiago/workspaces/crospi_test_ws/src/robot_drivers/trossen_widowx_driver_crospi/external/trossen_arm/demos/cpp/build
Consolidate compiler generated dependencies of target cartesian_external_effort
[  2%] Building CXX object CMakeFiles/cartesian_external_effort.dir/cartesian_external_effort.o
[  5%] Linking CXX executable cartesian_external_effort
/usr/bin/ld: cannot find -llibtrossen_arm: No such file or directory
/usr/bin/ld: note to link with /usr/local/lib/libtrossen_arm.a use -l:libtrossen_arm.a or rename it to liblibtrossen_arm.a
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/cartesian_external_effort.dir/build.make:97: cartesian_external_effort] Error 1
make[1]: *** [CMakeFiles/Makefile2:119: CMakeFiles/cartesian_external_effort.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

This is solved by adding:
```cmake
find_package(libtrossen_arm REQUIRED)
```

to the CMakeLists file of the demos. 

### Robot Model

xwai

### Driver Version

v1.9.1

### Firmware Version

v1.9.1

### System Info

```shell
Ubuntu 24
```

### Steps To Reproduce

_No response_

### Relevant Console Output

```shell

```

### Additional Info

_No response_

--- a/demos/cpp/CMakeLists.txt
+++ b/demos/cpp/CMakeLists.txt
@@ -1,3 +1,7 @@
+if(NOT TARGET libtrossen_arm)
+  find_package(libtrossen_arm REQUIRED)
+endif()
+
 include(FetchContent)
 
 FetchContent_Declare(

